motor_controller/motorbench-test-project/motorbench-test-project/rb_library/rb_pwm.c

```diff
@@ -162,3 +197,41 @@ bool RB_PWMCapBootstrapISRStep(RB_BOOTSTRAP *pBootstrap) {
     return bootstrapDone;
 }
 
+void RB_FixedFrequencySinePWM(uint16_t freqDivider)
+{
+    RunningStateCounter++;
+            
+    // Sine Frequency = 1/ (X*(1/20000)*297), where phaseIndex++ if RunningStateCounter >= X
+    // RPM = 120*f/52
+    if (RunningStateCounter >= freqDivider) // 9 works
+    {
+        RunningStateCounter = 0;
+
+        // retrieve sine() value
+        sineA = SineDutyCycle[phaseAIndex];
+        sineB = SineDutyCycle[phaseBIndex];
+        sineC = SineDutyCycle[phaseCIndex];
+
+        // set duties
+        MCC_PWM_DutyCycleSet(MOTOR1_PHASE_A,sineA);
+        MCC_PWM_DutyCycleSet(MOTOR1_PHASE_B,sineB);
+        MCC_PWM_DutyCycleSet(MOTOR1_PHASE_C,sineC);
+
+        // increment index of phases
+        phaseAIndex++;
```

Comment:
This is a lot of lines just to increment the index, would be better to use a function to increment all 3 like 'RB_PWMIncrementPhases', or the ternary operator.

Example with the ternary operator:
`phaseXIndex = (++phaseXIndex < (sin table size)) ? phaseXIndex : 0; `

Reply:
yeah